core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -227,6 +222,13 @@ private[spark] class Executor(
 
   metricsPoller.start()
```

Comment:
is there a reason we put this after metrics poller as well? I think I would rather init happen as soon as possible. Some people were using this for metrics initially.

Reply:
There is no strong reason so I'll put it just right after heartbeater.